Sources/EdgeNetworkHandlers/EdgeResponse.swift

```diff
@@ -17,7 +17,7 @@ import Foundation
 struct EdgeResponse: Codable {
 
     /// The request identifier associated with this response
-    let requestId: String?
+    let requestId: String
```

Comment:
Why was this change required?

Reply:
Marking the `requestId` as non-nil (required) was needed because if we would attempt to decode `Data` that represented an `EdgeEventError` to a `EdgeResponse` would end up with a non-nil `EdgeResponse` but with all nil properties.

Here is an example:
```swift
let edgeError = EdgeEventError(...)
let edgeErrorData = try! JSONEncoder().encode(edgeError)
let edgeResponse = try! JSONDecoder().decode(EdgeResponse.self, edgeErrorData)
XCTAssertNil(edgeResponse) 
```
Originally this assertion would have failed, it would have just initialized an empty `EdgeResponse` with all nil properties (since they were all optional). Now that we require there to be a `requestId` at the top level the decoding will fail, and return nil as we expect.

Reply:
This change essentially allow us to better match a response we'll get back from Kondcutor as `requestId` is a guaranteed field, thus allowing us to write cleaner code when it comes to decoding response vs error data.

Reply:
Can we fail the decoding when all properties are nil/not set?

Reply:
No, I looked into that. You could probably create a `isEmpty` flag that looks to see if all the properties are nil but I don't believe that's good practice in Swift. We should be marking all our decoded/encoded properties as nil or non-nil based on the API spec. However, some of the spec is a bit foggy to me as to whats optional/required.

AFAIK, there is no situation where we will receive a valid response (non-fatal) that doesn't contain a non-nil `requestId`. I'm having a hard time double-checking that as Konductors docs are giving back a 503.